### ISSUE TITLE
✅ 🐞 Bug Report: Elementor Pro unsets "div.widget_shopping_cart_content" cart fragment regardless if Mini Cart Template is disabled - [ED-10307]

### ISSUE THREAD

**Issue by hogash** [elementor/elementor#21836]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

In `ElementorPro\Modules\Woocommerce\Module` inside the `e_cart_count_fragments` method, the default WC Mini Cart fragments is removed with `unset( $fragments['div.widget_shopping_cart_content'] );` .

This happens regardless if the Mini Cart Template option is disabled or not.

The problem is that it causes incompatibilities with other themes/plugins which rely on this fragment.

### Steps to reproduce

Since it's likely only happening with other themes, i'm not sure of the steps to reproduce with Hello theme. Maybe with a default WP theme, not sure.

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [ ] I can reproduce this bug consistently following the steps above.

### System Info

I don't think it's relevant to this issue.

**Comment by nicholaszein** [comment]
Hello @hogash, 

Thank you for submitting your report! 🙏 

### ❌ While reviewing your report, we noticed it was missing some key pieces of information:

- 👣 **Steps to replicate:** Think of it as explaining to someone over the phone how to replicate your issue. They can't see the problem, so explain it in detail in a way that we can try to replicate your issue in our own environments. Be thorough.

- 💾 **System info:** Please include the System Info report as outlined in our [guidelines](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml). 
   - Get your System Info from your WordPress `Dashboard` > `Elementor` > `System info` [learn how to copy it here](https://elementor.com/help/hare-my-system-info/). We can't accept issues without it.

Without them, we simply can't help. 😕 

### 🧐 Please provide the missing information above so we can troubleshoot your issue faster and better.

Best regards,

Nick

**Comment by hogash** [comment]
Hi Nick,

This is the problem https://d.pr/i/Fe4JX0 and here is the fix https://d.pr/i/6kBfyC . If you want to pass this along the developers, fine, if not you can just close this issue.

PS: It's pretty annoying when getting this kind of reply when i specifically pointed out it's not the case. I understand there are thousands of issues, but some developer from Elementor's team needs to step in and also manage issues because they are very basic instead of this friction, they could be quickly handled.

Thanks!

**Comment by Squeezemind** [comment]
I can confirm this issue. Using a common wordpress theme for WooCommerce like shoptimizer and this stack:

WooCommerce 7.5.1
Shoptimizer 2.6.5
Elementor 3.12.0
Elementor Pro 3.11.7

All works fine. If I update Elementor Pro to version 3.12.0 the mini cart is not updated anymore. In both cases the mini cart option in Elementor’s settings is set to disabled.

I also recorded a video from where you can see the bug. 

https://user-images.githubusercontent.com/19249394/228886452-8ddc33e5-49d3-4048-851e-2526f4f3023b.mp4

**Comment by Squeezemind** [comment]
**To fix the issue** just edit the **elementor-pro/modules/woocommerce/module.php** and add the following code as the first instruction of the function **e_cart_count_fragments**:

`if( ! $this->use_mini_cart_template ) return $fragments;
`

As already suggested by @hogash

**Comment by accountsatelpengineering** [comment]
My website also has the issue of the cart missing in Shoptimizer.

Unfortunately I've lost sales because of it because people could not check out....

My website doesn't display the cart contents.



![Screenshot 2023-03-30 102952](https://user-images.githubusercontent.com/99732776/228907310-9dc63d61-c450-4cd3-8ad2-adca10cebe9d.jpg)

**Comment by accountsatelpengineering** [comment]
I've just rolled back to Elementor Pro 3.11.7 and it's back as it should be.

I will see if Elementor fix the code bug rather than put the suggested code in the .php file as i'm the business owner and not a DEV.

**Comment by nicholaszein** [comment]
Hi @hogash,

you know the guidelines. I can't send this issue forward without the System Info. Please provide the information required.

**Comment by Squeezemind** [comment]
Hi @nicholaszein ,
excuse me, but we are reporting a bug that afflicts a **paid version** of your plugin, with a ready-made fix, **all for free** and still expecting more work by us? But can't you simply release the patch asap since the bug affects e-commerce sites and doesn't allow customers to buy?

**Comment by marketingspectrumNL** [comment]
Guidelines are great, but a bit of flexibility when an experienced dev points out the issue AND hands out a fix would be even better. 

This is a basic php function that could be changed in minutes. I guess sometimes guidelines can get in your way. 
You know Marius, you know he knows what he's doing. If this was passed to the dev's we probably already had a fix for what is a critical issue for store owners.

**Comment by LiesBr** [comment]
Me too, shoptimizer theme.

**Comment by soundatech** [comment]
Same problem. All my websites with sidecart 👍 I change the code inside file, and no problem now!

**Comment by nicholaszein** [comment]
Hello guys...

This was passed to the QA team. But, guess what was missing? This means at least double the time to reproduce the issue, find out if it's something isolated, if it affects other features, and a whole lot of other tests that could be avoided if the guidelines where followed and respected.

We are trying to save time here. But without your collaboration that becomes impossible. So it will take more time to validate this issue, and fix it simply because the report was missing information that for us are crucial to help expedite the resolution of the issue.

This could have been solved a lot quicker if the user that reported the issue had respected everyone's time, including his own.

The issue is being investigated. We will release a patch if we find the bug, and when a solution is created.

Thank you everyone for collaborating with more information and feedback about your own installations. We really appreciate your kindness and patience.

I'll keep you posted ass soon as we have more info. 🙏

**Comment by loulemos** [comment]
Hey, just checking in to put my interest for this to be fixed, until then I can't update elementor pro on a few of our clients sites.

**Comment by accountsatelpengineering** [comment]
Update and warning to all: 

if you roll back to Elementor Pro 3.11.7, please go to the plugin page afterwards and '**disable auto updates**'.

I found out the hard way loosing two sales this morning as it updated itself and the cart disappeared again.

Regards, Ed

**Comment by Panark** [comment]
This is a very important and sirious bug that affected thousand of your customers. At least leave the formality in side and hands on to the problem. There a lost sales because of that and you have already ready the solution. Please open your mind and eys and say a BIG thank you to Marius for pointing out the solution to you... SAVING YOUR TIME!
Jesus... it is 2023 already and some "teams" are lost a decade before!!!

**Comment by nicholaszein** [comment]
Hello everyone, this issue is being investigated and a fix is underway!

We ask for patience until we can find a solution.

Thank you @hogash! 🙏

**Comment by domenasuperfashion** [comment]
I'm writing this just to comment: [nicholaszein] ...."We are trying to save time here. But without your collaboration that becomes impossible. So it will take more time to validate this issue, and fix it simply because the report was missing information that for us are crucial to help expedite the resolution of the issue. 
This could have been solved a lot quicker if the user that reported the issue had respected everyone's time, including his own.

the elementor team should also respect our time


I reported this error to the elementor team yesterday evening

if you offer them a staging site to reproduce the error.  what else could be faster and more detailed.

I have create staging and give them temporary link so that bug can be replicate. but since there is no valid license on my staging page (staging do not contain my site name so I can not use the same license), they did  nothing becouse of licenc: 

instead of solving the problem, from 8 am today, (when I got the first answer from help), until 2 pm  we argued about the license. and then it was the end of the working day and the matter remains unresolved over the weekend

My first  e-mail after I get help:  ...... I have upgrade first on staging copy and everything looks fine. On
staging I can not upgrade Pro version. After that I have upgrade also my production site including Elementor Pro. After upgrade my cart stop working. does not show the products in the cart. When I saw the problem  we restored my production site with backup. Now my staging and production site are working fine, but plugin is not updated.
I have made screenshot: In attach file you can see that product are added
to cart, but the cart is empty.

Temporary login link for staging: ..............

Please upgrade Pro version of Elementor on the staging site. an error should appear again.


Answer of Elementor: Kindly note we have accessed your site and we are seeing a Mismatch error. Please follow the troubleshooting steps to resolve this error so your license can be connected properly.

Let me know if you have any other questions.

My last e-mail: ...... And licence look fine. On my production site a problem occurred when I have upgrade to version 3.12..0. Now we have downgrade back with my buckup.. now you ask me to buy another license for use on the staging copy. My site is working now, but if I upgrade, the error will occur again. I asked you to upgrade the staging copy to version 3.12.0 with a temporary license and the error should reoccur. On the production side, I do not want to repeat the mistake, because in this case I would lose purchases. The error must be reproduced on staging and not on the live page.  If you insist on a valid license even for the staging copy, I will tell you it I will buy it, but within 30 days, as you allow, I will also request a refund of the purchase price. 
I was expecting help, but it feels like talking to a wall. It seems to me that you don't even read what I write to you.

**Comment by dailymarco** [comment]
Hello, I am suffering (Theme Shoptimizer) from the same problem. After I pasted the PHP code, it seems to be fixed for now. Please let me know when the update will come, many sites are affected.

Thank you very much!

**Comment by ygrealelement** [comment]
> Hello, I am suffering (Theme Shoptimizer) from the same problem. After I pasted the PHP code, it seems to be fixed for now. Please let me know when the update will come, many sites are affected.
> 
> Thank you very much!

Hello, I tried to paste the `if( ! $this->use_mini_cart_template ) return $fragments; ` however this doesn't fix it - maybe I am not pasting it correctly in the `e_cart_count` line. Could you please paste here the full command how it should look? That would help temporarily for us as well! 

Thank you in advance.

**Comment by imrelaszlo** [comment]
> > Hello, I am suffering (Theme Shoptimizer) from the same problem. After I pasted the PHP code, it seems to be fixed for now. Please let me know when the update will come, many sites are affected.
> > Thank you very much!
> 
> Hello, I tried to paste the `if( ! $this->use_mini_cart_template ) return $fragments; ` however this doesn't fix it - maybe I am not pasting it correctly in the `e_cart_count` line. Could you please paste here the full command how it should look? That would help temporarily for us as well!
> 
> Thank you in advance.

Please read this comment (and linked pages) carefully: https://github.com/elementor/elementor/issues/21836#issuecomment-1490317608

**Comment by Squeezemind** [comment]
> `if( ! $this->use_mini_cart_template ) return $fragments; `

Edit the **e_cart_count_fragments** function on **elementor-pro/modules/woocommerce/module.php** from:

```
public function e_cart_count_fragments( $fragments ) {
	$product_count = WC()->cart->get_cart_contents_count();

	$fragments['.elementor-menu-cart__toggle_button span.elementor-button-text'] = '<span class="elementor-button-text">' . WC()->cart->get_cart_subtotal() . '</span>';
	$fragments['.elementor-menu-cart__toggle_button span.elementor-button-icon-qty'] = '<span class="elementor-button-icon-qty" data-counter=' . $product_count . '>' . $product_count . '</span>';

	// Remove the default WC Mini Cart fragments as we will be doing our own AJAX call for this.
	unset( $fragments['div.widget_shopping_cart_content'] );

	return $fragments;
}
```

**to:**

```
public function e_cart_count_fragments( $fragments ) {
		
	if( ! $this->use_mini_cart_template ) return $fragments; 
	
	$product_count = WC()->cart->get_cart_contents_count();

	$fragments['.elementor-menu-cart__toggle_button span.elementor-button-text'] = '<span class="elementor-button-text">' . WC()->cart->get_cart_subtotal() . '</span>';
	$fragments['.elementor-menu-cart__toggle_button span.elementor-button-icon-qty'] = '<span class="elementor-button-icon-qty" data-counter=' . $product_count . '>' . $product_count . '</span>';

	// Remove the default WC Mini Cart fragments as we will be doing our own AJAX call for this.
	unset( $fragments['div.widget_shopping_cart_content'] );

	return $fragments;
}
```

**Comment by loulemos** [comment]
> Hello everyone, this issue is being investigated and a fix is underway!
> 
> We ask for patience until we can find a solution.
> 
> Thank you @hogash! 🙏

Hey! Appreciate it, we are currently using the 3.11.7 version and have not updated because of this, can you let us know when the fixed version is live so we can proceed?

**Comment by amyriounis** [comment]
@KingYes 
As a dev that supports several elementor-powered themes for thousands of clients, I have to say that is truly annoying how easy the Elementor team has been breaking mission-critical stuff of websites lately.

I'd advise instead of exercising all your strictness on people missing irrelevant system information when reporting issues for fixing your **proprietary** software, to try adding stricter rules for pushing production code that is affecting millions of sites.

I had to revert to WC's standard cart-fragments implementation to have the menu-cart operate normally again.

In case it helps other theme authors that don't have the luxury of updating files directly:

```php
function menu_cart_elementor_fix() {
	// Remove ElementorPro\Modules\Woocommerce\Module::menu_cart_fragments (fragments ajax handler).
	remove_all_actions( 'wp_ajax_elementor_menu_cart_fragments' );
	remove_all_actions( 'wp_ajax_nopriv_elementor_menu_cart_fragments' );

	// Remove ElementorPro\Modules\Woocommerce\Module::e_cart_count_fragments (fragments count ajax handler).
	function remove_e_cart_count_fragments_filter() {
		global $wp_filter;

		$filter = 'woocommerce_add_to_cart_fragments';
		$fn     = 'e_cart_count_fragments';

		if ( ! isset( $wp_filter[ $filter ] ) || empty( $wp_filter[ $filter ]->callbacks ) ) {
			return;
		}

		foreach( $wp_filter[ $filter ]->callbacks as $priority => $callbacks ) {
			foreach( $callbacks as $callback_key => $callback ) {
				if( strpos( $callback_key, $fn ) !== false ) {
					remove_filter( $filter, $callback_key, $priority );
				}
			}
		}
	}
	remove_e_cart_count_fragments_filter();

	// Add fragments count handler without causing issues.
	function cart_count_fragments( $fragments ) {
		$product_count = WC()->cart->get_cart_contents_count();

		$fragments['.elementor-menu-cart__toggle_button span.elementor-button-text'] = '<span class="elementor-button-text">' . WC()->cart->get_cart_subtotal() . '</span>';
		$fragments['.elementor-menu-cart__toggle_button span.elementor-button-icon-qty'] = '<span class="elementor-button-icon-qty" data-counter=' . $product_count . '>' . $product_count . '</span>';

		return $fragments;
	}
	add_filter( 'woocommerce_add_to_cart_fragments', 'cart_count_fragments' );
}
add_action( 'elementor/init', 'menu_cart_elementor_fix', 100 );
```

This is for aborting Elementor's custom Ajax call so there are no `Bad Request` errors (since we removed their handler on the backend):
```js
// TODO: Remove this when their fragments handling works without breaking WC.
function disableElementorMenuCartFragmentsAjaxHandler() {
    $.ajaxSetup({
        beforeSend: function ( xhr,settings ) {
            const isElementorMenuCartFragmentsHandler = ( settings.data &&  settings.data.indexOf( 'action=elementor_menu_cart_fragments' ) !== -1 );

            if ( isElementorMenuCartFragmentsHandler ) {
                xhr.abort();
            }
        }
    });
}
disableElementorMenuCartFragmentsAjaxHandler();
```

**Comment by WalletUpLLC** [comment]
Shame to elementor!!!! Shame on you! Always prompt to jump the dispute gun instead of addressing CRUCIAL issues. Worst, you're requiring some info completely IRRELEVANT to the issues affecting thousands!!!!! Smh!

Hogash Kudo!!! Thank you for the fix. We used **if( ! $this->use_mini_cart_template ) return $fragments;**  on line 823 of the plugins/elementor-pro/modules/woocommerce/module.php file and our issues were instantly resolved.

**Comment by waazaa** [comment]
> @KingYes As a dev that supports several elementor-powered themes for thousands of clients, I have to say that is truly annoying how easy the Elementor team has been breaking mission-critical stuff of websites lately.
> 
> I'd advise instead of exercising all your strictness on people missing irrelevant system information when reporting issues for fixing your **proprietary** software, to try adding stricter rules for pushing production code that is affecting millions of sites.
> 
> I had to revert to WC's standard cart-fragments implementation to have the menu-cart operate normally again.
> 
> In case it helps other theme authors that don't have the luxury of updating files directly:
> 
> ```
> function menu_cart_elementor_fix() {
> 	// Remove ElementorPro\Modules\Woocommerce\Module::menu_cart_fragments (fragments ajax handler).
> 	remove_all_actions( 'wp_ajax_elementor_menu_cart_fragments' );
> 	remove_all_actions( 'wp_ajax_nopriv_elementor_menu_cart_fragments' );
> 
> 	// Remove ElementorPro\Modules\Woocommerce\Module::e_cart_count_fragments (fragments count ajax handler).
> 	function remove_e_cart_count_fragments_filter() {
> 		global $wp_filter;
> 
> 		$filter = 'woocommerce_add_to_cart_fragments';
> 		$fn     = 'e_cart_count_fragments';
> 
> 		if ( ! isset( $wp_filter[ $filter ] ) || empty( $wp_filter[ $filter ]->callbacks ) ) {
> 			return;
> 		}
> 
> 		foreach( $wp_filter[ $filter ]->callbacks as $priority => $callbacks ) {
> 			foreach( $callbacks as $callback_key => $callback ) {
> 				if( strpos( $callback_key, $fn ) !== false ) {
> 					remove_filter( $filter, $callback_key, $priority );
> 				}
> 			}
> 		}
> 	}
> 	remove_e_cart_count_fragments_filter();
> 
> 	// Add fragments count handler without causing issues.
> 	function cart_count_fragments( $fragments ) {
> 		$product_count = WC()->cart->get_cart_contents_count();
> 
> 		$fragments['.elementor-menu-cart__toggle_button span.elementor-button-text'] = '<span class="elementor-button-text">' . WC()->cart->get_cart_subtotal() . '</span>';
> 		$fragments['.elementor-menu-cart__toggle_button span.elementor-button-icon-qty'] = '<span class="elementor-button-icon-qty" data-counter=' . $product_count . '>' . $product_count . '</span>';
> 
> 		return $fragments;
> 	}
> 	add_filter( 'woocommerce_add_to_cart_fragments', 'cart_count_fragments' );
> }
> add_action( 'elementor/init', 'menu_cart_elementor_fix', 100 );
> ```
> 
> This is for aborting Elementor's custom Ajax call so there are no `Bad Request` errors (since we removed their handler on the backend):
> 
> ```js
> // TODO: Remove this when their fragments handling works without breaking WC.
> function disableElementorMenuCartFragmentsAjaxHandler() {
>     $.ajaxSetup({
>         beforeSend: function ( xhr,settings ) {
>             const isElementorMenuCartFragmentsHandler = ( settings.data &&  settings.data.indexOf( 'action=elementor_menu_cart_fragments' ) !== -1 );
> 
>             if ( isElementorMenuCartFragmentsHandler ) {
>                 xhr.abort();
>             }
>         }
>     });
> }
> disableElementorMenuCartFragmentsAjaxHandler();
> ```

thank u!

**Comment by Rubenos12** [comment]
I'm sorry for commenting on this, but how can it take a week(!) to solve this, when you have the solution provided by @hogash ? This is not a small issue, it's an issue that makes conversions drop lik crazy, what results in revenue loss. Please, get it fixed, without having to put snippets in the code.

**Comment by Stas0238** [comment]
@Panark 
It is "funny" to mention that they had updates every 2-5 days for their "tweaks" https://prnt.sc/2Qoxz9Pm3oZB but for real "bugs" they want customers patience for longer time.
"GOOD" idea Elementor developers !

**Comment by Fredb57** [comment]
Took my about 3 minutes to fix the problem just copy paste the hogash solution.
I am not a real developer just someone who could not understand why sales are so low last week.
Thanks Elementor for (not) being flexible.

**Comment by scottybo** [comment]
Just echoing we're seeing this issue across three client sites

**Comment by nicholaszein** [comment]
Hello everyone, I wanted to give you an update to this issue:

A patch is underway, and if all test go as planned it should be fixed in the next patch planned for next Sunday.

Thank you for your patience!

**Comment by imrelaszlo** [comment]
> A patch is underway, and if all test go as planned it should be fixed in the next patch planned for next Sunday. 

Next Sunday? Why don't you release an intermediate version? There is no need for ChatGPT and similar nonsense, it should not be deal with. This software is a WordPress page builder, not an AI backed bullshit. Eliminating critical errors like this should be the highest priority. Guys, after such attitude, I think we should consider using other software instead of Elementor...

**Comment by ygrealelement** [comment]
> > A patch is underway, and if all test go as planned it should be fixed in the next patch planned for next Sunday.
> 
> Next Sunday? Why don't you release an intermediate version? There is no need for ChatGPT and similar nonsense, it should not be deal with. This software is a WordPress page builder, not an AI backed bullshit. Eliminating critical errors like this should be the highest priority. Guys, after such attitude, I think we should consider using other software instead of Elementor...

Indeed, It doesn't make sense, however once a company gets big enough it starts to pay less and less attention to crucial parts. 

A small update focusing precisely on a major issue as this one could have already been provided. However it seems the hype-train with this ChapGPT is of course more important for marketing value (only temporarily).

**Comment by raabmar** [comment]
Glad I found this post, but I spent (wasted) another few hours, so even though I am happy that Elementor is on the fast track now (compared to the last years), the amount of hours that must be invested is just insane these days. Please try at least to test it with several themes before releasing it (I read that Woodmart, Ocean WP, and even Storefront are all affected)

**Comment by dailymarco** [comment]
How the Elementor team dealt with this serious problem is a scandal and extremely flawed for a "premium provider". sad.

**Comment by CodeConnectPtyLtd** [comment]
I can't count the number of sales our shop lost because of this.  User decides to buy, clicks add to cart and gets no response form the interface == no sale == lost revenue.  How many other shops lost THOUSANDS in revenue over this period without even knowing?  I discover this Easter Friday because a customer REALLY wanted to purchase and contacted us.  I then spend my Easter Sunday fixing the SH!T that you lot broke.

**Comment by popov593** [comment]
This should really be fixed urgently - it is easily fix-able and is affecting an enormous amount of websites.

**Comment by Stas0238** [comment]
I have an idea that Elementor just wants to make their "brand" name to be on each lips. And they does not think that most of those "lips' owners" are talking about Elementor's CRITICAL BUG they left without fixing. Yeahh, Elementor - let's keep this crutial bug for 2 weeks more and then release a fix with some "super needed" tweaks that 95% of clients just don't need.
Support this answer if you share my thought.

**Comment by nicholaszein** [comment]
⚠️ Update [This is important]:

For those of you who used @hogash's solution provided [here](https://github.com/elementor/elementor/issues/21836#issuecomment-1490317608), and @Squeezemind's solution provided [here](https://github.com/elementor/elementor/issues/21836#issuecomment-1494028125), or any other solution for that matter, keep in mind it will cause the issue that was reported and solved in #19312.

So if you've made any changes, be aware that it will affect the cache. We were able to pinpoint this side effect thanks to the **System Info** provided by everyone who shared it. This is why sharing <ins>**ALL**</ins> the information required is so important.

We have already fixed this issue, and it is planned to be released in the next patch on Monday with the correct fix.

If you can't wait until then, you can rollback to version 3.11.7 which is the latest stable and secure version.

Thank you to everyone who shared the required information, and most importantly for being patient. 🙏

**Comment by doronwolf** [comment]
Hi everyone 3.12.2 pro is out with the fix💪!
thank you all for your patience

**Comment by dailymarco** [comment]
> 

Thank you very much for fixing such a serious bug with such breathtaking speed after "only" 2 weeks. We will be forever grateful to you.......

**Comment by nexondo** [comment]
Oh that amazing! Thank you so much for fixing this issue so fast!

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

As @doronwolf mentioned yesterday,

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro **v3.12.2**! 🥳 

Thank you very much to everyone who contributed information to solve this issue, as well as giving your feedback about our procedures. 🙏 

We are constantly working on improving our communication, and your feedback is extremely valuable! We understand sometimes it might be frustrating to wait or to have to share all the information we request. But by doing that you are helping us identify possible related issues, and take the necessary time to fix them properly to avoid further complications.

Nonetheless, every single comment here was crucial for fixing this issue, and we are grateful for your contribution!

✅ Feel free to update your plugin to the new version!

Cheers 🥂

**Comment by nicholaszein** [comment]
Just a heads up:

If you're still experiencing issues after the update, and you are using the Mini Cart from your theme or another plugin, please make sure the WooCommerce Mini Cart Template option is turned off in your `Dashboard` > `Elementor` > `Settings` > `Advanced` Page. This option should only be active if you are using the Elementor Menu Cart widget.

**Comment by aloaax** [comment]
not working for me shoptimizer theme 

![image](https://user-images.githubusercontent.com/130428751/231054599-2a7df204-2b38-406b-aaa0-1b65d0435e70.png)

**Comment by aloaax** [comment]
> update

not working with me

**Comment by soundatech** [comment]
Same way!

**Comment by loulemos** [comment]
Good morning, @nicholaszein seems like it's not working for some users. I'm also using the Shoptimizer theme for a few clients so I'm reluctant to update. Would do it on a staging site but alas.

**Comment by Fredb57** [comment]
> Just a heads up:
> 
> If you're still experiencing issues after the update, and you are using the Mini Cart from your theme or another plugin, please make sure the WooCommerce Mini Cart Template option is turned off in your `Dashboard` > `Elementor` > `Settings` > `Advanced` Page. This option should only be active if you are using the Elementor Menu Cart widget.

Do I miss something?
![Elementor-settings-advanced](https://user-images.githubusercontent.com/77838957/231126705-02d984ba-effd-488e-9142-13c692ec14f3.jpg)

**Comment by domenasuperfashion** [comment]
Dashboard > Elementor > Settings >Integrations  (... WooCommerce)

**Comment by nicholaszein** [comment]
We tested and it should work if the WooCommerce Mini Cart Template integration is deactivated. Make sure to set it to Inactive and Save Changes. 

It's important to make sure you have cleared the cache. Especially from server side.

**Comment by nicholaszein** [comment]
Hello @haniobaid if you think this is a bug, please submit a new issue. This one is for a different topic and has already been closed.

Follow our guidelines to submit a new issue, and make sure you have cleared the cache. No changes were made to any styles, so your issue is likely caused by not clearing all levels of cache after updating.

**Comment by nicholaszein** [comment]
Hello everyone,

After further investigation, we concluded that some of the new features added to the Menu Cart widget were still causing conflicts beyond the ones we patched in Elementor Pro version 3.12.2. For this reason, we decided it would be better to revert the changes made in version 3.12 and rework the implementation of the new features to avoid conflicts that were still reported in some cases.

We want to extend our **apologies** to everyone who was affected by these bugs and report that we have released a patch for Elementor Pro version **3.12.3** that should fix all the remaining issues.

Thank you to everyone who contributed, and most importantly for being so patient. 🙏 

You can now update to version 3.12.3 and use Elementor Pro without needing to roll back.

If you are still experiencing any issues after updating to this version, clearing the cache, and regenerating CSS & Data, please let us know in the comments down below.

I will be monitoring this threat for the next weeks until the launch of version 3.13.

----

From our official statement posted in the Elementor Global Community:

>📣 Important Update: Menu Cart Widget
> We have an important update to share with you regarding the Menu Cart widget in our upcoming Elementor 3.12.3 Pro patch. You might recall the handy feature that allowed you to add custom icons for the Remove Item button. Unfortunately, we've decided to roll back this feature for now. 😓
> With this patch installed, you'll only be able to use the default icons provided by Elementor for the Remove Item button. This change resulted from our commitment to prioritizing stability and reliability, as we faced some unexpected hiccups and glitches with the custom icons feature.
> We understand that this might be a disappointment for those of you who were excited to personalize your icons. However, please know that our team is constantly working on new features and improvements to make your Elementor experience even better.
> If you are experiencing any issues with the Mini Cart in your theme, we suggest that you roll back to version 3.11.7 until the patch is released.
> We genuinely appreciate your support, understanding, and patience as we refine and optimize Elementor. If you have any questions or concerns regarding this update, please don't hesitate to comment or contact our support team.
> THANK YOU FOR BEING A PART OF OUR COMMUNITY AND FOR CONTINUING TO CREATE GREAT WEBSITES WITH ELEMENTOR. STAY CREATIVE, STAY AWESOME, AND KEEP ON BUILDING! 🚀
https://www.facebook.com/groups/Elementors/posts/5833067723485632/

Kind regards 🙏

**Comment by waazaa** [comment]
Thank u Nicholas, I think you just miss to delete the "elementor-lightbox" class in the "elementor-menu-cart__container", because it shows me the grey transparent bg of the lightboxes. Everything else works perfect. I use the hello theme, and jetengine listings.

**Comment by haniobaid** [comment]
Just to confirm the plugin conflict I mentioned earlier was indeed resolved on updating to 3.12.3
Thanks for the quick turnaround!
